guides/deep-dives/field-editors.md

```diff
@@ -1,7 +1,119 @@
+    {{outlet}}
+  {{/squishable-container}}
+{{/cardstack-edges}}
+```
+
+For someone to see the editing panel, the must have a valid user session and the right grants that allow them to edit content. You can learn more about authorization, authentication, and mock users for testing in [Grants](../../data/grants/).
+
+## Built-in field editors
+
+Field editors are components that handle user input.
+Cardstack provides some built-in content editor components.
+
+When a field is set to certain `core-types` in a Card's schema, some editors get automatically applied:
+
+- Boolean: `field-editors/boolean-editor` is automatically provided for `@cardstack/core-types::boolean`
+- Date: `field-editors/date-editor` is automatically provided for `@cardstack/core-types::date`
+- Integer: `field-editors/integer-editor` is automatically provided for `@cardstack/core-types::integer`
+- String: `field-editors/string-editor` is automatically provided for `@cardstack/core-types::string`
+
+Other built-in editors should be set for each field in the Card's schema: 
+
+- Datetime: use `field-editors/datetime-editor`
+- Dropdown: use 
+    - `field-editors/dropdown-choices-editor`
+    - `field-editors/dropdown-multi-select-editor`
+    - `field-editors/dropdown-search-editor`
+    - `field-editors/dropdown-search-multi-select-editor`
+
+For example, this dropdown menu lets someone assign a `category` grouping to a photo Card:
+
+```js
+    factory.addResource('fields', 'category').withAttributes({
+      fieldType: '@cardstack/core-types::belongs-to',
+      editorOptions: { displayFieldName: 'name' },
+      editorComponent: 'field-editors/dropdown-choices-editor'
+})
+```
+
+## Passing options to a field editor
+
+Field editors can take options using the `editorOptions` property in a Card's schema for a specific field:
+
+```js
+factory.addResource('fields', 'is-admin-user').withAttributes({
+  fieldType: '@cardstack/core-types::boolean',
+  editorOptions: { style: 'switch' }
+})
+```
+
+To see which options are availble for each of the built-in field editors, see the [core types documentation](https://github.com/cardstack/cardstack/tree/master/packages/core-types).
+
+## Creating your own field editor component
+
+Imagine that you want the editing field to have some input validation behavior, or it is more complex than the bult-in editors.
+You can create your own field editor components for a Card.
+
+From within the directory for a Card, run:
+
+```bash
+ember generate component field-editors/my-editor-name.
+```
+
+This will generate three files:
+
+- a Template at `card-name/addon/templates/field-editors/my-editor-name.hbs` which handles display
+- a JavaScript file at `card-name/addon/components/field-editors/my-editor-name.js`, where you can add interaction and logic
+- a second JavaScript file at `card-name/addon/components/field-editors/my-editor-name.js` with some boilerplate that reexports the component, which you will not need to edit.
```

Comment:
Shouldn't this be the js file in the `app` folder?

Reply:
Yes, thanks!